salt/modules/disk.py

```diff
@@ -464,11 +479,18 @@ def fstype(device):
     if salt.utils.which('df'):
         # the fstype was not set on the block device, so inspect the filesystem
         # itself for its type
-        df_out = __salt__['cmd.run']('df -T {0}'.format(device)).splitlines()
-        if len(df_out) > 1:
-            fs_type = df_out[1]
-            if fs_type:
-                return fs_type
+        if __grains__['kernel'] == 'AIX' and os.stat('/usr/sysv/bin/df'):
+            df_out = __salt__['cmd.run']('/usr/sysv/bin/df -n {0}'.format(device)).split()
+            if len(df_out) > 2:
+                fs_type = df_out[2]
+                if fs_type:
```

Comment:
I can't quite follow this. You're splitting a string into a list of component parts. Under what condition would this be true? We already fall through and return `''` in the event of an empty string so I'm confused about what condition you're guarding against here.